dfdaemon/proxy/proxy.go

```diff
@@ -330,14 +330,19 @@ func (proxy *Proxy) handleHTTPS(w http.ResponseWriter, r *http.Request) {
 	// We have to wait until the connection is closed
 	wg := sync.WaitGroup{}
 	wg.Add(1)
-	http.Serve(&singleUseListener{&customCloseConn{sConn, wg.Done}}, rp)
+	if err := http.Serve(&singleUseListener{&customCloseConn{sConn, wg.Done}}, rp); err != nil {
+		logrus.Errorf("failed to accepts incoming HTTP connections: %v", err)
+	}
 	wg.Wait()
 }
 
-func copyAndClose(dst io.WriteCloser, src io.ReadCloser) {
-	io.Copy(dst, src)
+func copyAndClose(dst io.WriteCloser, src io.ReadCloser) error {
+	if _, err := io.Copy(dst, src); err != nil {
+		return err
```

Comment:
If you return here, `Close` on both `dst` and `src` will not be called

Reply:
The change here is because of @Starnop  suggestion, maybe you can discuss it with him.

Reply:
@hwdef just introduce `defer to make it`. I agree with @inoc603